### ISSUE TITLE
NodeStageVolume is called before NodeUnstageVolume returns success

### ISSUE THREAD

**Issue by rohitChaware** [kubernetes/kubernetes#102486]
#### What happened:
We have a StatefulSet pod using a CSI volume.
After rolling restart, the pod got scheduled on the 'same node' as where it was running before.
The NodeUnstageVolume was called for the CSI volume and the `umount` operation was running.
Before the NodeUnstageVolume returned success, NodeStageVolume was called for the volume.
The `umount` operation of the NodeUnstageVolume deleted the staging directory: `/var/lib/kubelet/plugins/kubernetes.io/csi/pv/csi-8de1ce2a-ffbf-4446-a3a5-117798676551/globalmount`  after NodeStageVolume returned Success.

#### What you expected to happen:
NodeStageVolume to be issued only after NodeUnstageVolume returns success.

#### How to reproduce it (as minimally and precisely as possible):
Above scenario happens intermittently; seems to be a race condition. 

#### Environment:
- Kubernetes version (use `kubectl version`): 1.17.13
- external-attacher v2.1.0

**Comment by dguendisch** [comment]
This sounds like a duplicate of https://github.com/kubernetes/kubernetes/issues/100182
